File: update_gnb.py
import os
import re

files = [
    'index.html',
    'brand.html',
    'career.html',
    'product.html',
    'product-reset-mist.html',
    'product-restore-serum.html',
    'product-recover-balm.html'
]

html_gnb = """<!-- ===================== GNB START ===================== -->
<nav id="gnb" style="
  position: sticky;
  top: 0;
  z-index: 1000;
  background: rgba(242,240,232,0.92);
  backdrop-filter: blur(16px);
  -webkit-backdrop-filter: blur(16px);
  border-bottom: 0.5px solid #c8d4b0;
  height: 64px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 48px;
  box-sizing: border-box;
">

  <!-- 로고 -->
  <a href="/index.html" style="
    font-family: 'Pretendard', sans-serif;
    font-size: 20px;
    font-weight: 500;
    font-style: italic;
    color: #4a5e2f;
    text-decoration: none;
    flex-shrink: 0;
  ">GenYou</a>

  <!-- PC 메뉴 -->
  <div class="gnb-pc-menu" style="
    display: flex;
    align-items: center;
    gap: 4px;
  ">

    <!-- BRAND -->
    <a href="/brand.html" class="gnb-link">BRAND</a>

    <!-- PRODUCT (드롭다운) -->
    <div class="gnb-dropdown-wrap">
      <a href="/product.html" class="gnb-link gnb-has-dropdown">
        PRODUCT
        <svg width="10" height="10" viewBox="0 0 10 10" fill="none"
             style="margin-left:3px; vertical-align:middle; transition:transform .2s;"
             class="gnb-arrow">
          <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor"
                stroke-width="1.2" stroke-linecap="round"/>
        </svg>
      </a>

      <!-- 드롭다운 패널 -->
      <div class="gnb-dropdown">
        <div style="
          font-family: 'Pretendard', sans-serif;
          font-size: 10px;
          font-weight: 500;
          color: #7a9a3f;
          letter-spacing: 0.08em;
          text-transform: uppercase;
          padding: 10px 14px 6px;
        ">3SecFix™ 라인업</div>

        <a href="/product.html" class="gnb-dropdown-item">
          <span style="font-weight:500;">전체 제품 보기</span>
          <span class="gnb-dropdown-sub">Reset Mist · Restore Serum · Recover Balm</span>
        </a>

        <div style="height:0.5px; background:#e8e4da; margin:4px 14px;"></div>

        <a href="/product-reset-mist.html" class="gnb-dropdown-item">
          <span style="font-weight:500;">Reset Mist</span>
          <span class="gnb-dropdown-sub">칙! — 유수분 최적 밸런싱</span>
        </a>

        <a href="/product-restore-serum.html" class="gnb-dropdown-item">
          <span style="font-weight:500;">Restore Serum</span>
          <span class="gnb-dropdown-sub">톡! — 촘촘히 채워주는 수분 밀도</span>
        </a>

        <a href="/product-recover-balm.html" class="gnb-dropdown-item">
          <span style="font-weight:500;">Recover Cream Balm</span>
          <span class="gnb-dropdown-sub">쓱! — 가볍지만 쫀쫀한 마지막 보호막</span>
        </a>
      </div>
    </div>

    <!-- COMMUNITY -->
    <a href="/brand.html#community" class="gnb-link">COMMUNITY</a>

    <!-- CAREER -->
    <a href="/career.html" class="gnb-link">CAREER</a>

  </div>

  <!-- 진단하기 버튼 + 햄버거 -->
  <div style="display:flex; align-items:center; gap:12px; flex-shrink:0;">
    <button onclick="openDiagnosis()" class="gnb-cta-btn"
      aria-label="피부 진단 시작하기">
      진단하기
    </button>
    <button class="gnb-hamburger" onclick="openMobileMenu()"
      aria-label="메뉴 열기">
      <span></span>
      <span></span>
      <span></span>
    </button>
  </div>

</nav>

<!-- 모바일 드로어 -->
<div id="mobile-drawer" style="
  display: none;
  position: fixed;
  inset: 0;
  z-index: 2000;
  background: #f2f0e8;
  flex-direction: column;
  padding: 24px 24px 40px;
  box-sizing: border-box;
  overflow-y: auto;
">

  <!-- 드로어 헤더 -->
  <div style="display:flex; justify-content:space-between; align-items:center;
              margin-bottom:40px;">
    <a href="/index.html" style="
      font-family: 'Pretendard', sans-serif;
      font-size: 20px; font-weight: 500;
      font-style: italic; color: #4a5e2f;
      text-decoration: none;">GenYou</a>
    <button onclick="closeMobileMenu()" style="
      background: none; border: none; cursor: pointer;
      font-size: 24px; color: #6b6b6b; line-height: 1;
      padding: 4px;">✕</button>
  </div>

  <!-- 드로어 메뉴 목록 -->
  <div style="display:flex; flex-direction:column; gap:0;">

    <a href="/brand.html" onclick="closeMobileMenu()"
       class="drawer-link">BRAND</a>

    <!-- PRODUCT 탭 (클릭으로 하위 펼침) -->
    <div>
      <button onclick="toggleProductMenu()" class="drawer-link drawer-toggle"
        id="product-toggle">
        PRODUCT
        <svg width="14" height="14" viewBox="0 0 14 14" fill="none"
             id="product-arrow" style="transition:transform .25s;">
          <path d="M3 5L7 9L11 5" stroke="#2e2e2e"
                stroke-width="1.2" stroke-linecap="round"/>
        </svg>
      </button>

      <!-- PRODUCT 하위 메뉴 -->
      <div id="product-submenu" style="
        display: none;
        flex-direction: column;
        background: #EAF3DE;
        border-radius: 12px;
        margin: 4px 0 8px;
        overflow: hidden;
      ">
        <a href="/product.html" onclick="closeMobileMenu()"
           class="drawer-sub-link">
          <span style="font-weight:500;">전체 제품 보기</span>
          <span style="font-size:11px; color:#6b6b6b; display:block; margin-top:1px;">
            3SecFix™ 라인업
          </span>
        </a>
        <div style="height:0.5px; background:#c8d4b0; margin:0 16px;"></div>
        <a href="/product-reset-mist.html" onclick="closeMobileMenu()"
           class="drawer-sub-link">
          <span style="font-weight:500;">Reset Mist</span>
          <span style="font-size:11px; color:#6b6b6b; display:block; margin-top:1px;">
            칙! — 유수분 최적 밸런싱
          </span>
        </a>
        <a href="/product-restore-serum.html" onclick="closeMobileMenu()"
           class="drawer-sub-link">
          <span style="font-weight:500;">Restore Serum</span>
          <span style="font-size:11px; color:#6b6b6b; display:block; margin-top:1px;">
            톡! — 촘촘히 채워주는 수분 밀도
          </span>
        </a>
        <a href="/product-recover-balm.html" onclick="closeMobileMenu()"
           class="drawer-sub-link">
          <span style="font-weight:500;">Recover Cream Balm</span>
          <span style="font-size:11px; color:#6b6b6b; display:block; margin-top:1px;">
            쓱! — 가볍지만 쫀쫀한 마지막 보호막
          </span>
        </a>
      </div>
    </div>

    <a href="/brand.html#community" onclick="closeMobileMenu()"
       class="drawer-link">COMMUNITY</a>

    <a href="/career.html" onclick="closeMobileMenu()"
       class="drawer-link">CAREER</a>

  </div>

  <!-- 드로어 하단 진단하기 버튼 -->
  <div style="margin-top:auto; padding-top:40px;">
    <button onclick="openDiagnosis(); closeMobileMenu();" style="
      width: 100%; height: 56px;
      background: #4a5e2f; color: #fff; border: none;
      border-radius: 16px; cursor: pointer;
      font-family: 'Pretendard', sans-serif;
      font-size: 16px; font-weight: 500;
    ">진단하기 시작하기</button>
  </div>

</div>
<!-- ===================== GNB END ===================== -->"""

# User specifically requested these exact lines
css_extra = """
.gnb-pc-menu { display: flex; }
"""

for fname in files:
    with open(fname, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Check if .gnb-pc-menu { display: flex; } exists in style
    if '.gnb-pc-menu { display: flex; }' not in content:
        # insert it right before the media queries
        if '/* 반응형 */' in content:
            content = content.replace('/* 반응형 */', css_extra + '\n/* 반응형 */')
        else:
            content = content.replace('</style>', css_extra + '\n</style>')
            
    with open(fname, 'w', encoding='utf-8') as f:
        f.write(content)
        
    print(f"Updated {fname}")

File: update_gnb.js
const fs = require('fs');
const path = require('path');

const htmlGnb = `<!-- ===== GNB START ===== -->
<nav id="gnb">
  <!-- 로고 -->
  <a href="/index.html" class="gnb-logo">GenYou</a>

  <!-- PC 4개 메뉴 -->
  <div class="gnb-pc-menu">

    <!-- OUR STORY -->
    <div class="gnb-drop-wrap">
      <button class="gnb-link gnb-drop-btn">OUR STORY</button>
      <div class="gnb-dropdown">
        <a href="/our-story-brand.html" class="gnb-drop-item">
          Brand Story<span class="gnb-drop-sub">젠유랩의 시작</span>
        </a>
        <a href="/our-story-3secfix.html" class="gnb-drop-item">
          3SecFix™<span class="gnb-drop-sub">성분과 원칙</span>
        </a>
        <a href="/our-story-philosophy.html" class="gnb-drop-item">
          Philosophy<span class="gnb-drop-sub">내버려 두는 것의 미학</span>
        </a>
      </div>
    </div>

    <!-- SHOP -->
    <div class="gnb-drop-wrap">
      <button class="gnb-link gnb-drop-btn">SHOP</button>
      <div class="gnb-dropdown gnb-mega-menu">
        <div class="gnb-accordion">
          <!-- 1. 3SecFix™ -->
          <div class="gnb-acc-item" tabindex="0" aria-expanded="false">
            <button class="gnb-acc-trigger" type="button">
              3SecFix™
              <svg width="10" height="10" viewBox="0 0 10 10" fill="none" class="gnb-acc-arrow">
                <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor" stroke-width="1.2" stroke-linecap="round"/>
              </svg>
            </button>
            <div class="gnb-acc-content">
              <div class="gnb-acc-inner">
                <a href="/product.html?category=3step-3secfix" class="gnb-drop-item">
                  3SecFix™ 3Step<span class="gnb-drop-sub">3초 3단계 풀 패키지</span>
                </a>
                <a href="/product-reset-mist.html" class="gnb-drop-item">
                  리셋 베일 토너<span class="gnb-drop-sub">칙! — 유수분 밸런싱</span>
                </a>
                <a href="/product-restore-serum.html" class="gnb-drop-item">
                  리스토어 세럼<span class="gnb-drop-sub">톡! — 수분 밀도</span>
                </a>
                <a href="/product-recover-balm.html" class="gnb-drop-item">
                  리커버 크림 밤<span class="gnb-drop-sub">쓱! — 마지막 보호막</span>
                </a>
              </div>
            </div>
          </div>

          <!-- 2. 올리브 부스터 라인 -->
          <div class="gnb-acc-item" tabindex="0" aria-expanded="false">
            <button class="gnb-acc-trigger" type="button">
              올리브 부스터 라인
              <svg width="10" height="10" viewBox="0 0 10 10" fill="none" class="gnb-acc-arrow">
                <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor" stroke-width="1.2" stroke-linecap="round"/>
              </svg>
            </button>
            <div class="gnb-acc-content">
              <div class="gnb-acc-inner">
                <a href="/product.html?category=olive-glow-booster" class="gnb-drop-item">
                  올리브 글로우 부스터<span class="gnb-drop-sub">광채 피부의 시작</span>
                </a>
              </div>
            </div>
          </div>

          <!-- 3. 썬,바디,헤어 라인 -->
          <div class="gnb-acc-item" tabindex="0" aria-expanded="false">
            <button class="gnb-acc-trigger" type="button">
              썬,바디,헤어 라인
              <svg width="10" height="10" viewBox="0 0 10 10" fill="none" class="gnb-acc-arrow">
                <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor" stroke-width="1.2" stroke-linecap="round"/>
              </svg>
            </button>
            <div class="gnb-acc-content">
              <div class="gnb-acc-inner">
                <a href="/product.html?category=sun-care" class="gnb-drop-item">
                  썬케어<span class="gnb-drop-sub">피부 보호 선스크린</span>
                </a>
                <a href="/product.html?category=body-care" class="gnb-drop-item">
                  바디 케어<span class="gnb-drop-sub">바디 진정 및 보습</span>
                </a>
                <a href="/product.html?category=hair-care" class="gnb-drop-item">
                  헤어 케어<span class="gnb-drop-sub">두피 및 모발 영양</span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <!-- EXPERIENCE -->
    <div class="gnb-drop-wrap">
      <button class="gnb-link gnb-drop-btn">EXPERIENCE</button>
      <div class="gnb-dropdown">
        <a href="/experience-review.html" class="gnb-drop-item">
          Review<span class="gnb-drop-sub">실제 후기 모음</span>
        </a>
        <a href="/experience-sns.html" class="gnb-drop-item">
          SNS<span class="gnb-drop-sub">인스타그램 · 유튜브</span>
        </a>
        <a href="/experience-event.html" class="gnb-drop-item">
          Event<span class="gnb-drop-sub">진행중인 이벤트</span>
        </a>
      </div>
    </div>

    <!-- SUPPORT -->
    <div class="gnb-drop-wrap">
      <button class="gnb-link gnb-drop-btn">SUPPORT</button>
      <div class="gnb-dropdown">
        <a href="/support-faq.html" class="gnb-drop-item">
          FAQ<span class="gnb-drop-sub">자주 묻는 질문</span>
        </a>
        <a href="/support-inquiry.html" class="gnb-drop-item">
          1:1 Inquiry<span class="gnb-drop-sub">개인 문의</span>
        </a>
        <a href="/career.html" class="gnb-drop-item">
          Join crew<span class="gnb-drop-sub">함께할 사람을 찾습니다</span>
        </a>
      </div>
    </div>

  </div><!-- /gnb-pc-menu -->

  <!-- 우측: 삼단바 버튼 하나 -->
  <button class="gnb-hamburger" onclick="openUtilMenu()" aria-label="메뉴 열기">
    <span></span><span></span><span></span>
  </button>

</nav>

<!-- ===== 유틸리티 드로어 (삼단바 클릭 시) ===== -->
<div id="util-drawer">
  <div class="util-inner">
    <div class="util-header">
      <a href="/index.html" class="gnb-logo" style="color:#4a5e2f">GenYou</a>
      <button onclick="closeUtilMenu()" class="util-close">✕</button>
    </div>

    <!-- 유틸리티 링크 (Login, Join, Cart, Mypage) -->
    <div class="util-links">
      <a href="/login.html" class="util-link">
        <span class="util-icon">
          <svg width="20" height="20" viewBox="0 0 20 20" fill="none">
            <circle cx="10" cy="7" r="3.5" stroke="#2e2e2e" stroke-width="1.2"/>
            <path d="M3 17c0-3.3 3.1-6 7-6s7 2.7 7 6" stroke="#2e2e2e" stroke-width="1.2" stroke-linecap="round"/>
          </svg>
        </span>
        <span class="util-link-text">Login</span>
        <span class="util-arrow">→</span>
      </a>
      <a href="/join.html" class="util-link">
        <span class="util-icon">
          <svg width="20" height="20" viewBox="0 0 20 20" fill="none">
            <circle cx="10" cy="7" r="3.5" stroke="#2e2e2e" stroke-width="1.2"/>
            <path d="M3 17c0-3.3 3.1-6 7-6s7 2.7 7 6" stroke="#2e2e2e" stroke-width="1.2" stroke-linecap="round"/>
            <path d="M16 4v4M14 6h4" stroke="#4a5e2f" stroke-width="1.2" stroke-linecap="round"/>
          </svg>
        </span>
        <span class="util-link-text">Join Crew</span>
        <span class="util-arrow">→</span>
      </a>
      <a href="/cart.html" class="util-link">
        <span class="util-icon">
          <svg width="20" height="20" viewBox="0 0 20 20" fill="none">
            <path d="M2 3h2l2.5 9h9l2-6H6" stroke="#2e2e2e" stroke-width="1.2" stroke-linecap="round" stroke-linejoin="round"/>
            <circle cx="9" cy="16.5" r="1.2" fill="#2e2e2e"/>
            <circle cx="14" cy="16.5" r="1.2" fill="#2e2e2e"/>
          </svg>
        </span>
        <span class="util-link-text">Cart</span>
        <span class="util-arrow">→</span>
      </a>
      <a href="/mypage.html" class="util-link">
        <span class="util-icon">
          <svg width="20" height="20" viewBox="0 0 20 20" fill="none">
            <rect x="3" y="3" width="14" height="14" rx="3" stroke="#2e2e2e" stroke-width="1.2"/>
            <path d="M7 8h6M7 11h4" stroke="#2e2e2e" stroke-width="1.2" stroke-linecap="round"/>
          </svg>
        </span>
        <span class="util-link-text">Mypage</span>
        <span class="util-arrow">→</span>
      </a>
    </div>

    <!-- 구분선 -->
    <div class="util-divider"></div>

    <!-- 모바일에서만: GNB 메뉴 -->
    <div class="util-nav-mobile">
      <div class="util-nav-section">
        <div class="util-nav-title">OUR STORY</div>
        <a href="/our-story-brand.html" onclick="closeUtilMenu()">Brand Story</a>
        <a href="/our-story-3secfix.html" onclick="closeUtilMenu()">3SecFix™</a>
        <a href="/our-story-philosophy.html" onclick="closeUtilMenu()">Philosophy</a>
      </div>
      <div class="util-nav-section">
        <div class="util-nav-title">SHOP</div>
        <div class="gnb-accordion">
          <!-- 1. 3SecFix™ -->
          <div class="gnb-acc-item" tabindex="0" aria-expanded="false">
            <button class="gnb-acc-trigger" type="button">
              3SecFix™
              <svg width="10" height="10" viewBox="0 0 10 10" fill="none" class="gnb-acc-arrow">
                <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor" stroke-width="1.2" stroke-linecap="round"/>
              </svg>
            </button>
            <div class="gnb-acc-content">
              <div class="gnb-acc-inner">
                <a href="/product.html?category=3step-3secfix" onclick="closeUtilMenu()">3step 3SecFix™</a>
                <a href="/product-reset-mist.html" onclick="closeUtilMenu()">리셋 베일 토너</a>
                <a href="/product-restore-serum.html" onclick="closeUtilMenu()">리스토어 세럼</a>
                <a href="/product-recover-balm.html" onclick="closeUtilMenu()">리커버 크림 밤</a>
              </div>
            </div>
          </div>

          <!-- 2. 올리브 부스터 라인 -->
          <div class="gnb-acc-item" tabindex="0" aria-expanded="false">
            <button class="gnb-acc-trigger" type="button">
              올리브 부스터 라인
              <svg width="10" height="10" viewBox="0 0 10 10" fill="none" class="gnb-acc-arrow">
                <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor" stroke-width="1.2" stroke-linecap="round"/>
              </svg>
            </button>
            <div class="gnb-acc-content">
              <div class="gnb-acc-inner">
                <a href="/product.html?category=olive-glow-booster" onclick="closeUtilMenu()">올리브 글로우 부스터</a>
              </div>
            </div>
          </div>

          <!-- 3. 썬,바디,헤어 라인 -->
          <div class="gnb-acc-item" tabindex="0" aria-expanded="false">
            <button class="gnb-acc-trigger" type="button">
              썬,바디,헤어 라인
              <svg width="10" height="10" viewBox="0 0 10 10" fill="none" class="gnb-acc-arrow">
                <path d="M2 3.5L5 6.5L8 3.5" stroke="currentColor" stroke-width="1.2" stroke-linecap="round"/>
              </svg>
            </button>
            <div class="gnb-acc-content">
              <div class="gnb-acc-inner">
                <a href="/product.html?category=sun-care" onclick="closeUtilMenu()">썬케어</a>
                <a href="/product.html?category=body-care" onclick="closeUtilMenu()">바디 케어</a>
                <a href="/product.html?category=hair-care" onclick="closeUtilMenu()">헤어 케어</a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="util-nav-section">
        <div class="util-nav-title">EXPERIENCE</div>
        <a href="/experience-review.html" onclick="closeUtilMenu()">Review</a>
        <a href="/experience-sns.html" onclick="closeUtilMenu()">SNS</a>
        <a href="/experience-event.html" onclick="closeUtilMenu()">Event</a>
      </div>
      <div class="util-nav-section">
        <div class="util-nav-title">SUPPORT</div>
        <a href="/support-faq.html" onclick="closeUtilMenu()">FAQ</a>
        <a href="/support-inquiry.html" onclick="closeUtilMenu()">1:1 Inquiry</a>
        <a href="/career.html" onclick="closeUtilMenu()">Join crew</a>
      </div>
    </div>

  </div>
</div>
<!-- ===== GNB END ===== -->`;

// Regex pattern to search for the GNB comment blocks
const gnbRegex = /<!--\s*={3,}\s*GNB START\s*={3,}\s*-->[\s\S]*?<!--\s*={3,}\s*GNB END\s*={3,}\s*-->/gi;

const files = fs.readdirSync(__dirname);
let updatedCount = 0;

files.forEach(file => {
  if (file.endsWith('.html') && fs.statSync(file).isFile()) {
    try {
      const filePath = path.join(__dirname, file);
      let content = fs.readFileSync(filePath, 'utf8');

      if (gnbRegex.test(content)) {
        // Replace matching GNB blocks
        content = content.replace(gnbRegex, htmlGnb);
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Updated GNB in: ${file}`);
        updatedCount++;
      } else {
        console.log(`No GNB block found in: ${file}`);
      }
    } catch (e) {
      console.error(`Error processing ${file}:`, e);
    }
  }
});

console.log(`\nSuccessfully updated GNB in ${updatedCount} files.`);
